zanata/zanata-platform
# feat(ZNTA-2318): Editor Review / Reject Translation

JIRA issue URL: https://zanata.atlassian.net/browse/ZNTA-2318

Modification of the RejectTranslationModal frontend container and backend rest API in progress to implement this feature.

## QA
The alpha editor now has the approve and reject options for each text flow.
Clicking reject or selecting it with the key combination ( CTRL + ⇧ + S R ) should open a modal with criteria and priority dropdown fields, and a comment text field.
The priority field selection has not been implemented in the backend as the details around this option are still being discussed #672 .
The comment is also not required to submit a rejection.
Behavior when there are no criteria to select is also still being determined, this PR can be tested with existing criteria (added through frontend admin panel -> review criteria)
After adding a review comment it should be observable in the GWT editors translation history.
![rejecting](https://user-images.githubusercontent.com/7971187/35902521-3a41d55a-0c27-11e8-98c9-c8ff0a7709e0.png)

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zanata/zanata-platform/688)
